workflow/scripts/spia.R

```diff
@@ -46,5 +46,33 @@ if(nrow(sig_genes) == 0) {
     setwd(olddir)
 
     file.copy(file.path(t, "SPIAPerturbationPlots.pdf"), snakemake@output[["plots"]])
+    pathway_names <- db[res$Name]
+    pathway_names <- db[res$Name]
+    path_ids <- as.matrix(lapply(pathway_names@entries, slot, "id"))
+    if (length(path_ids) > 0 ) {
+            path_ids_data_frame <-
+        data.frame(Ids = matrix(unlist(path_ids),
+            nrow = length(path_ids), byrow = TRUE))
+    final_res <- cbind(res,
+        Ids = path_ids_data_frame$Ids)
+    res_reorder <- dplyr::select(final_res, Name, Status,
+        pGFdr, tA, pSize, NDE, pNDE, pGFWER, pPERT, pG, Ids)
+    names(res_reorder)[names(res_reorder) == "pGFWER"] <- "Combined Bonferroni p-values"
+    names(res_reorder)[names(res_reorder) == "pGFdr"] <- "Combined FDR"
+    names(res_reorder)[names(res_reorder) == "tA"] <- "total perturbation accumulation"
+    names(res_reorder)[names(res_reorder) == "pSize"] <- "number of genes on the pathway"
+    names(res_reorder)[names(res_reorder) == "NDE"] <- "number of DE genes per pathway"
+    names(res_reorder)[names(res_reorder) == "pG"] <- "Combined p-value"
+    names(res_reorder)[names(res_reorder) == "pPERT"] <- "p-value to observe a total accumulation"
+    names(res_reorder)[names(res_reorder) == "pNDE"] <- "p-value for at least NDE genes"
+    write_tsv(res_reorder, snakemake@output[["table"]])
+    }else {
```

Comment:
please ensure that this is properly formatted (e.g. using styleR or checking the R style guide manually).

Reply:
Noted. Updated